packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1419,6 +1422,24 @@ abstract class BuildableElement extends Element {
     markNeedsBuild();
   }
 
+  bool _hadDependenciesWhenDeactivated = false;
```

Comment:
I was just thinking we could follow the pattern in RenderObject.attach which uses the render tree dirty bits in this way already.